nano/secure/ledger_set_any.cpp

```diff
@@ -0,0 +1,213 @@
+{
+	auto block = block_get (transaction, ledger.any.account_head (transaction, account_a));
+	if (!block)
+	{
+		return std::nullopt;
+	}
+	return block->balance ();
+}
+
+auto nano::ledger_set_any::account_begin (secure::transaction const & transaction) const -> account_iterator
+{
+	return account_lower_bound (transaction, 0);
+}
+
+auto nano::ledger_set_any::account_end () const -> account_iterator
+{
+	return account_iterator{};
+}
+
+std::optional<nano::account_info> nano::ledger_set_any::account_get (secure::transaction const & transaction, nano::account const & account) const
+{
+	return ledger.store.account.get (transaction, account);
+}
+
+nano::block_hash nano::ledger_set_any::account_head (secure::transaction const & transaction, nano::account const & account) const
+{
+	auto info = account_get (transaction, account);
+	if (!info)
+	{
+		return 0;
+	}
+	return info.value ().head;
+}
+
+uint64_t nano::ledger_set_any::account_height (secure::transaction const & transaction, nano::account const & account) const
+{
+	auto head_l = account_head (transaction, account);
+	if (head_l.is_zero ())
+	{
+		return 0;
+	}
+	auto block = block_get (transaction, head_l);
+	release_assert (block); // Head block must be in ledger
+	return block->sideband ().height;
+}
+
+auto nano::ledger_set_any::account_lower_bound (secure::transaction const & transaction, nano::account const & account) const -> account_iterator
+{
+	auto disk = ledger.store.account.begin (transaction, account);
+	if (disk == ledger.store.account.end ())
+	{
+		return account_iterator{};
+	}
+	return account_iterator{ transaction, *this, *disk };
+}
+
+auto nano::ledger_set_any::account_upper_bound (secure::transaction const & transaction, nano::account const & account) const -> account_iterator
+{
+	return account_lower_bound (transaction, account.number () + 1);
+}
+
+std::optional<nano::account> nano::ledger_set_any::block_account (secure::transaction const & transaction, nano::block_hash const & hash) const
+{
+	auto block_l = block_get (transaction, hash);
+	if (!block_l)
+	{
+		return std::nullopt;
+	}
+	return block_l->account ();
+}
+
+std::optional<nano::amount> nano::ledger_set_any::block_amount (secure::transaction const & transaction, nano::block_hash const & hash) const
+{
+	auto block_l = block_get (transaction, hash);
+	if (!block_l)
+	{
+		return std::nullopt;
+	}
+	auto block_balance = block_l->balance ();
+	if (block_l->previous ().is_zero ())
+	{
+		return block_balance.number ();
+	}
+	auto previous_balance = this->block_balance (transaction, block_l->previous ());
+	if (!previous_balance)
+	{
+		return std::nullopt;
+	}
+	return block_balance > previous_balance.value () ? block_balance.number () - previous_balance.value ().number () : previous_balance.value ().number () - block_balance.number ();
+}
+
+// Balance for account containing hash
```

Comment:
This comment seems wrong, it can be removed since the name of the function is now obvious.